models.ts

```diff
@@ -1,17 +1,54 @@
 import * as jsonfile from "jsonfile";
-
-// no modificar estas propiedades, agregar todas las que quieras
+import * as find from "lodash/find";
+import * as includes from "lodash/includes";
 class Peli {
   id: number;
   title: string;
   tags: string[];
 }
 
-class PelisCollection {
+class PelisCollection extends Peli {
   getAll(): Promise<Peli[]> {
-    return jsonfile("...laRutaDelArchivo").then(() => {
-      // la respuesta de la promesa
-      return [];
+    return jsonfile.readFile("./pelis.json").then((res) => {
+      return res;
+    });
+  }
+  getById(id: number): Promise<any> {
+    return this.getAll().then((archivo) => {
+      return find(archivo, { id: id });
+    });
+  }
+  search(options: any): Promise<any> {
+    return this.getAll().then((archivo) => {
+      var pelisEncontradas = archivo;
+      if (options.title) {
+        pelisEncontradas = pelisEncontradas.filter((peli) => {
+          return includes(
+            peli.title.toLocaleLowerCase(),
+            options.title.toLocaleLowerCase()
+          );
+        });
+      }
+      if (options.tag) {
+        pelisEncontradas = pelisEncontradas.filter((peli) => {
+          return peli.tags.find((tag) => {
+            return tag == options.tag;
+          });
+        });
+      }
+      return pelisEncontradas;
+    });
+  }
+  add(peli: Peli): Promise<any> {
+    return this.getAll().then((archivo) => {
+      const existe = find(archivo, { id: peli.id });
```

Comment:
todo esto lo hacés tal cual en this.getById podés reutilizarlo.

Reply:
ah bien, no me di cuenta, tenes razón

Reply:
revisé este punto y no lo modifiqué porque "existe" es una variable de tipo boolean que se necesita usar en ese momento y si uso el método getById me devuelve una promesa y primero se resuelve el "if" antes que la promesa